Failed to download Quiltflower

```
java.util.concurrent.CompletionException: java.io.FileNotFoundException: https://maven.quiltmc.org/repository/snapshot/org/quiltmc/quiltflower/maven-metadata.xml
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:331)
	at java.base/java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:1099)
	at java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2235)
	at net.earthcomputer.quiltflowerintellij.QuiltflowerState.downloadQuiltflower(QuiltflowerState.kt:119)
	at net.earthcomputer.quiltflowerintellij.QuiltflowerState.initialize(QuiltflowerState.kt:72)
	at net.earthcomputer.quiltflowerintellij.QuiltflowerPreloadingActivity.preload(QuiltflowerPreloadingActivity.kt:8)
	at com.intellij.idea.ApplicationLoader$executePreloadActivity$1.run(ApplicationLoader.kt:418)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:624)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:698)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:646)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:623)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:66)
	at com.intellij.idea.ApplicationLoader.executePreloadActivity(ApplicationLoader.kt:408)
	at com.intellij.idea.ApplicationLoader.access$executePreloadActivity(ApplicationLoader.kt:1)
	at com.intellij.idea.ApplicationLoader$executePreloadActivities$2.run(ApplicationLoader.kt:473)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.io.FileNotFoundException: https://maven.quiltmc.org/repository/snapshot/org/quiltmc/quiltflower/maven-metadata.xml
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1920)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1520)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:250)
	at net.earthcomputer.quiltflowerintellij.QuiltflowerState.downloadQuiltflowerVersions$download(QuiltflowerState.kt:195)
	at net.earthcomputer.quiltflowerintellij.QuiltflowerState.access$downloadQuiltflowerVersions$download(QuiltflowerState.kt:25)
	at net.earthcomputer.quiltflowerintellij.QuiltflowerState$downloadQuiltflowerVersions$1.invoke(QuiltflowerState.kt:214)
	at net.earthcomputer.quiltflowerintellij.QuiltflowerState$downloadQuiltflowerVersions$1.invoke(QuiltflowerState.kt:211)
	at net.earthcomputer.quiltflowerintellij.UtilKt.runOnBackgroundThreadWithProgress(util.kt:20)
	at net.earthcomputer.quiltflowerintellij.QuiltflowerState.downloadQuiltflowerVersions(QuiltflowerState.kt:211)
	at net.earthcomputer.quiltflowerintellij.QuiltflowerState.initialize(QuiltflowerState.kt:71)
	... 16 more
```

--- a/src/main/kotlin/net/earthcomputer/quiltflowerintellij/QuiltflowerState.kt
+++ b/src/main/kotlin/net/earthcomputer/quiltflowerintellij/QuiltflowerState.kt
@@ -12,6 +12,7 @@ import com.intellij.util.io.readText
 import com.intellij.util.text.SemVer
 import com.intellij.util.xmlb.XmlSerializerUtil
 import com.intellij.util.xmlb.annotations.Transient
+import java.io.FileNotFoundException
 import java.net.HttpURLConnection
 import java.net.URL
 import java.net.URLClassLoader
@@ -192,16 +193,21 @@ class QuiltflowerState : PersistentStateComponent<QuiltflowerState> {
         fun download(baseUrl: String): Pair<SemVer?, List<SemVer>> {
             val latestVersion: SemVer?
             val allVersions: List<SemVer>
-            URL(baseUrl + "maven-metadata.xml").openConnection().getInputStream().use { inputStream ->
-                val element = JDOMUtil.load(inputStream)
-                if (element.name != "metadata") {
-                    throw IllegalStateException("Invalid metadata file")
+            try {
+                URL(baseUrl + "maven-metadata.xml").openConnection().getInputStream().use { inputStream ->
+                    val element = JDOMUtil.load(inputStream)
+                    if (element.name != "metadata") {
+                        throw IllegalStateException("Invalid metadata file")
+                    }
+                    val versioning = element.getChild("versioning") ?: throw IllegalStateException("Invalid metadata file")
+                    latestVersion = SemVer.parseFromText(versioning.getChild("latest")?.text)
+                    allVersions = versioning.getChild("versions")?.children?.mapNotNull {
+                        SemVer.parseFromText(it.text)
+                    } ?: emptyList()
                 }
-                val versioning = element.getChild("versioning") ?: throw IllegalStateException("Invalid metadata file")
-                latestVersion = SemVer.parseFromText(versioning.getChild("latest")?.text)
-                allVersions = versioning.getChild("versions")?.children?.mapNotNull {
-                    SemVer.parseFromText(it.text)
-                } ?: emptyList()
+            } catch (e: FileNotFoundException) {
+                LOGGER.warn("${baseUrl}maven-metadata.xml could not be found.")
+                return null to emptyList()
             }
             return latestVersion to allVersions
         }

